### DISCUSSION TITLE
Is it possible to allow redundant fields to be inserted when using mongodb?

### DISCUSSION THREAD

**Discussion by lokiikoll** [prisma/prisma#13910] [Q&A]
I am using mongodb.
I need to store a lot of fields, but I don't want all fields in my prisma model definition.
Because the data I crawl is complex, and some field structures are not fixed.

**Comment by austincrim** [comment]
Hey @heijinggh 👋,

You can always define a field as `Json` type and store anything in it, it doesn't have to be defined in the Prisma schema.
```prisma
model User {
  id  Int @id
  settings Json // any JSON can go in here!
}

```


You will at least need a `model` for every MongoDB collection that you are interacting with, though.

**Reply by lokiikoll** [reply]
Thanks for your help.
I mean there are many unknown properties at the root.
```js
model User {
  id  Int @id
  settings Json // any JSON can go in here!
  fieldA // JSON or String or any other type
  fieldB
  ....
}
```

**Reply by lokiikoll** [reply]
> Hey @heijinggh 👋,
> 
> You can always define a field as `Json` type and store anything in it, it doesn't have to be defined in the Prisma schema.
> 
> ```prisma
> model User {
>   id  Int @id
>   settings Json // any JSON can go in here!
> }
> ```
> 
> You will at least need a `model` for every MongoDB collection that you are interacting with, though.

There are some properties, I'm not sure about their names and types, but I need to save them. And they may be on the root node.

**Reply by austincrim** [reply]
You will at least need to have a known root node and then you can put any arbitrary JSON underneath that property.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
